xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/cluster/FlushTask.java

```diff
@@ -4,21 +4,41 @@
 import java.io.ObjectInput;
 import java.io.ObjectOutput;
 
-import org.jivesoftware.openfire.pubsub.PubSubPersistenceManager;
+import org.jivesoftware.openfire.pubsub.CachingPubsubPersistenceProvider;
+import org.jivesoftware.openfire.pubsub.Node;
+import org.jivesoftware.openfire.pubsub.PubSubPersistenceProvider;
+import org.jivesoftware.openfire.pubsub.PubSubPersistenceProviderManager;
 import org.jivesoftware.util.cache.ClusterTask;
+import org.jivesoftware.util.cache.ExternalizableUtil;
 
 
 public class FlushTask implements ClusterTask<Void>
 {
+	private Node.UniqueIdentifier uniqueIdentifier;
+
+	public FlushTask( Node.UniqueIdentifier uniqueIdentifier )
+	{
+		this.uniqueIdentifier = uniqueIdentifier;
+	}
+
     public FlushTask()
     {
+		this.uniqueIdentifier = null;
     }
 
     @Override
     public void run()
     {
-        PubSubPersistenceManager.flushPendingItems(false); // just this member
-    }
+		final PubSubPersistenceProvider provider = PubSubPersistenceProviderManager.getInstance().getProvider();
+		if ( provider instanceof CachingPubsubPersistenceProvider )
+		{
+			if ( uniqueIdentifier != null ) {
+				((CachingPubsubPersistenceProvider) provider).flushPendingItems( uniqueIdentifier, false ); // just this member
+			} else {
+				((CachingPubsubPersistenceProvider) provider).flushPendingItems( false ); // just this member
+			}
```

Comment:
The comment `just this member` appears in both branches - is that right?

Reply:
Yes, it refers to the 'false' argument - it limits the action to the local cluster member.